### PR TITLE
Fix: [UI] BottomSheet

### PR DESCRIPTION
**Pull request by markom-povio**
Improvements and PR fixes for [UI] BottomSheet
---
- Added separate file for WizardDataSource.swift
- Changed year in Copyright header
- Added private extension in WizardContentView
- Change name of the component from "Wizard" to the "BottomSheet"
- BottomSheet: Added outside tap gesture to dismiss (swipe to dismiss is already available with modal presentation)
- BottomSheetContentView: addHelperGestureRecognizer to skip "tapToDismiss" on content view
- Added README file
- Rename for BottomSheetAnimators file

> Note: Simple Demo project is on separate branch: [BottomSheet Demo](https://github.com/poviolabs/PovioKit/tree/chore/demo_bottom_sheet)

### CODE REVIEW

**Comment by tonikocjan** [comment]
@markom-povio there is a lot of stuff here that I think should not be part of this PR? Can you check.

File: Resources/Controllers/BottomSheet/README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+#  BottomSheet
```
Comment (tonikocjan) [inline Resources/Controllers/BottomSheet/README.md:1]: Why the name change? [Wizard](https://docs.microsoft.com/en-us/windows/win32/uxguide/win-wizards) is a common name used for a multi-step user interface.

File: Resources/Controllers/BottomSheet/README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+A UI component that effectively replaces and extends native UIActionSheet.
```
Comment (tonikocjan) [inline Resources/Controllers/BottomSheet/README.md:3]: I think `Wizard` is a bit of an overkill as a replacement for UIActionSheet as it prepares a lot of stuff regarding animations between different steps and so on.

File: Resources/Controllers/BottomSheet/README.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
+#  BottomSheet
+
+A UI component that effectively replaces and extends native UIActionSheet.  
+
+## Overview  
+
+BottomSheet has the following characteristics:
+- dynamically expandable based on the content height
+- swipe down gesture to dismiss
+- tap gesture behind the content to dismiss
```
Comment (tonikocjan) [inline Resources/Controllers/BottomSheet/README.md:10]: These (the gestures) should be left for the users to implement/integrate in any way they want. The Wizard itself should be as self-contained as possible. It can actually be used as an embedded view controller inside another to provide context for a wizard(multi step)-like UI.

File: Resources/Controllers/BottomSheet/README.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+- **LazyStep**: 
+```swift
+typealias LazyStep = () -> BottomSheetStep
+```
+- **AnimatorFactory**: *Enum that holds predefined transition Animations*
+```swift
+enum AnimatorFactory
+```
+
+### Default animations:
```
Comment (tonikocjan) [inline Resources/Controllers/BottomSheet/README.md:40]: I would change wording from `Default` to `Provided` or `Predefined` or smth similar.

File: Resources/Controllers/BottomSheet/README.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+- slideInSlideOut
+- fadeIn
+- fadeOut
+- fadeInFadeOut
+- popInPopOut  
+
+Besides these predefined animations, it is possible to combine two or more animations with the available `composite` function, but also to create custom animation with return type: ```typealias Animator = (UIView, UIView) -> Void```
```
Comment (tonikocjan) [inline Resources/Controllers/BottomSheet/README.md:47]: `Besides these predefined animations, it is possible ...` -> `Furthermore, it is possible ...`.

**Review by tonikocjan — COMMENTED** [review]
Thanks for your work on this @markom-povio. Left a couple of comments to look at.

**Comment by markom-povio** [comment]
@tonikocjan I will reply for most of the comments here because they are more or less related:

> Why the name change? [Wizard](https://docs.microsoft.com/en-us/windows/win32/uxguide/win-wizards) is a common name used for a multi-step user interface.  

I do not mind having the *Wizard* name, the idea was to align with the parent issue [BottomSheet](https://github.com/poviolabs/PovioKit/issues/155)

> I think Wizard is a bit of an overkill as a replacement for UIActionSheet as it prepares a lot of stuff regarding animations between different steps and so on.  

Yes, probably we do not need animations (or even the Steps and transition between them) if we want simple UIActionSheet replacement. But we have here a nice solution that covers both scenarios if we want to present simple or more complex bottomSheet UI.  

> These (the gestures) should be left for the users to implement/integrate in any way they want. The Wizard itself should be as self-contained as possible. It can actually be used as an embedded view controller inside another to provide context for a wizard(multi step)-like UI.  

The swipe gesture we already have out of the box with presenting modal BottomSheetViewController, but the tap gesture was one of the requirements in the parent issue [BottomSheet](https://github.com/poviolabs/PovioKit/issues/155)

Basically, most of the questions are on point here, but the solution is depending of the idea behind the component.  
Do we need this complexity of having the Steps, animations and tap gesture and should we rename the component back to the *Wizard* name so it better reflects the behaviour (I think this should be done if all the logic stays as it is)?

@borut-t what do you think?

**Comment by borut-t** [comment]
Thanks @tonikocjan and @markom-povio for digging in. I think we've made a small mistake here since we've actually joined two components together where they should be separated. A wizard and a dialogue. I've discuss this with Toni this morning and we would like to create both instead. @markom-povio I'll update the ticket wording so it will better suit it's function as it is currently. Sorry for this, but I was a bit lame here not clearly check the existing implementation from Toni.
